### DISCUSSION TITLE
Build or developer documentation (how to build from scratch)

### DISCUSSION THREAD

**Discussion by jdhoek** [osmandapp/OsmAnd#14402] [Q&A] [answered]
I would like to contribute missing OSM tags to OsmAnd's default styles. Is there documentation somewhere that focusses on building OsmAnd from source? In particular: _if I change something in [OsmAnd-resources](https://github.com/osmandapp/OsmAnd-resources) with the intent of submitting a PR, what then is needed to get an `apk` I can install on Android to test my changes?_ (And: is that the usual way to test such changes?)

-----

As usual with Java based projects I had a go at simply running the build in [OsmAnd](https://github.com/osmandapp/OsmAnd), but I get the following with Java 11:

```
$ ./gradlew

FAILURE: Build failed with an exception.

* Where:
Build file '/home/freaknl/Dev/osm/OsmAnd/OsmAnd/build.gradle' line: 1

* What went wrong:
A problem occurred evaluating project ':OsmAnd'.
> Failed to apply plugin 'com.android.internal.application'.
   > Your project contains 2 or more modules with the same identification net.osmand:OsmAnd
     at ":" and ":OsmAnd".
     You must use different identification (either name or group) for each modules.

```

I'm sure I'm missing something obvious. Am I overlooking a README containing build prerequisites?

**Comment by vshcherb** [comment, marked as answer]
Here is documentation - https://osmand.net/docs/technical/build-osmand/how-to-compile-the-android-version/

**Reply by jdhoek** [reply]
Thanks. I didn't expect the developer documentation to be included in the user docs.
